### DISCUSSION TITLE
Multiobjective optimization with constraints, (MO)TPE vs NSGA-II, etc.

### DISCUSSION THREAD

**Discussion by themagpipers** [optuna/optuna#5259] [Q&A] [answered]
I am new to Optuna and I would like to have some clarifications.
If I understand well, for a multiobjective problem, by default Optuna uses the NSGA-II algorithm. In my case, I can already solve my problem with NSGA-II (see Platypus or Pymoo librairies). I would really like to use a surrogate model to solve the problem, this is why I went for Optuna.

Therefore, I have specified `sampler = optuna.samplers.TPESampler()` and `study = optuna.create_study(directions=['minimize', 'minimize'], sampler = sampler)` to use the Tree-Parzen estimator algorithm, which seems to work fine in my case. I still have to figure out how to deal with the constraints (I return a large number when a constraint is violated, but this is bad since this doesn't give TPE any information on how badly the constraints are violated). 

I see in the source code that I should use MOTPE for multiobjective problems, which is my use-case. However I also see that this algorithm is deprecated and will be removed. What should I replace it with? Is the current TPE OK? In the source code, it is claimed that MOTPE is better than TPE... I am confused.

**Comment by nzw0301** [comment, marked as answer]
As in the released note https://github.com/optuna/optuna/releases/tag/v2.9.0, MOTPESampler  should be replaced with TPESampler.

**Comment by themagpipers** [comment]
Thanks a lot, I see.

Another question I have is: when I use the TPESampler, I see that there is an n_startup_trials which uses a random sampling before the TPE is used. But this random sampling is actually uniform rather than a LHS algorithm, right? It would make sense to use a LHS instead, is there an easy way to modify the choice of the random sampler?

**Reply by nzw0301** [reply]
From the definition of the TPESampler class, replacing https://github.com/optuna/optuna/blob/master/optuna/samplers/_tpe/sampler.py#L307 with your sampler can be a way to do so. Defining sampler is described in https://optuna.readthedocs.io/en/stable/tutorial/20_recipes/005_user_defined_sampler.html.
